sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -431,19 +431,21 @@ public void dispose() {
     }
 
     Mono<Void> closeAsync(AmqpShutdownSignal shutdownSignal) {
-        addShutdownSignal(logger.atInfo(), shutdownSignal).log("Disposing of ReactorConnection.");
-        final Sinks.EmitResult result = shutdownSignalSink.tryEmitValue(shutdownSignal);
-
-        if (result.isFailure()) {
-            // It's possible that another one was already emitted, so it's all good.
-            addShutdownSignal(logger.atInfo(), shutdownSignal)
-                .addKeyValue(EMIT_RESULT_KEY, result)
-                .log("Unable to emit shutdown signal.");
-        }
+        final Mono<Void> emitShutDownSignalOperation = Mono.fromRunnable(() -> {
+            addShutdownSignal(logger.atInfo(), shutdownSignal).log("Disposing of ReactorConnection.");
+            final Sinks.EmitResult result = shutdownSignalSink.tryEmitValue(shutdownSignal);
+
+            if (result.isFailure()) {
+                // It's possible that another one was already emitted, so it's all good.
+                addShutdownSignal(logger.atInfo(), shutdownSignal)
+                    .addKeyValue(EMIT_RESULT_KEY, result)
+                    .log("Unable to emit shutdown signal.");
+            }
+        });
 
         final Mono<Void> cbsCloseOperation;
         if (cbsChannelProcessor != null) {
-            cbsCloseOperation = cbsChannelProcessor.flatMap(channel -> channel.closeAsync());
+            cbsCloseOperation = Mono.fromRunnable(() -> cbsChannelProcessor.dispose());
```

Comment:
Because if we want to close cbs channel here, we couldn't use `repeat().takeUtilOther(shutdownSignal)` to stop request new cbs channel after connection closed.
Here I try to use `AmqpChannelProcess.dispose()` to mark the processor as disposed and close channels. But this need to double check the reason why use `flatMap(channel -> channel.closeAsync())` previously.

Reply:
1. Use `cbsCloseOperation = cbsChannelProcessor.flatMap(channel -> channel.closeAsync())` will return `Mono<Void> closeOperationWithTimeout`, so cbsCloseOperation will block until Detach frame is received from remote link.

![cbs-closing-frame-logic1](https://user-images.githubusercontent.com/97088315/162147201-7ffe61cf-da73-470a-b95c-99d4917e4bf7.png)

2. Use `cbsCloseOperation = Mono.fromRunnable(() -> cbsChannelProcessor.dispose())` only invoke the `link.close()` on `reactor-executor` thread. So it just trigger to send out the detach frame.

![cbs-closing-frame-logic2](https://user-images.githubusercontent.com/97088315/162147229-e5c88819-1fd7-4446-9b97-b93513559354.png)

Reply:
We should choose the option 1 to make sure CBS node is closed before closing sessions.